arkane/orca.py

```diff
@@ -0,0 +1,231 @@
+# copy of this software and associated documentation files (the 'Software'),  #
+# to deal in the Software without restriction, including without limitation   #
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,    #
+# and/or sell copies of the Software, and to permit persons to whom the       #
+# Software is furnished to do so, subject to the following conditions:        #
+#                                                                             #
+# The above copyright notice and this permission notice shall be included in  #
+# all copies or substantial portions of the Software.                         #
+#                                                                             #
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,    #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+Arkane Orca module
+Used to parse Orca output files
+"""
+
+import logging
+
+import rmgpy.constants as constants
+
+from arkane.common import get_element_mass
+from arkane.log import Log
+from arkane.exceptions import LogError
+
+
+################################################################################
+
+
+class OrcaLog(Log):
+    """
+    Represent an output file from Orca. The attribute `path` refers to the
+    location on disk of the Orca output file of interest. Methods are provided
+    to extract a variety of information into Arkane classes and/or NumPy
+    arrays.
+    """
+
+    def __init__(self, path):
+        super(OrcaLog, self).__init__(path)
+
+    def get_number_of_atoms(self):
+        """
+        Return the number of atoms in the molecular configuration used in
+        the Orca output file.
+        """
+        natoms = 0
+
+        with open(self.path, 'r') as f:
+            line = f.readline()
+            while line != '' and natoms == 0:
+                # Automatically determine the number of atoms
+                if 'CARTESIAN COORDINATES (ANGSTROEM)' in line and natoms == 0:
+                    for i in range(2):
+                        line = f.readline()
+
+                    while '---------------------------------' not in line:
+                        natoms += 1
+                        line = f.readline()
+                        if not line.strip():
+                            return natoms
```

Comment:
I'm not 100% certain, but I think that if we return within an `with open` block, we should close the file before returning

Reply:
I didn't see a response to this comment